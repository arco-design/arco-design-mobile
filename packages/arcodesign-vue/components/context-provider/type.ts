export interface ContextInstallOptions {
    /**
     * 组件名称前缀
     * @en Component prefix
     * @default "Arco"
     */
    componentPrefix?: string;
}

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
}
