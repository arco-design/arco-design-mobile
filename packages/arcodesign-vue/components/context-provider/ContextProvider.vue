<script lang="ts" setup>
import { provide } from 'vue';
import { GlobalContextParams } from './type';
import { configProviderInjectionKey, defaultContext } from './utils';

const props = withDefaults(defineProps<GlobalContextParams>(), defaultContext);
provide(configProviderInjectionKey, props);
</script>
<template>
    <slot />
</template>
