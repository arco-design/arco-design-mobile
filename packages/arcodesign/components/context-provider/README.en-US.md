### Others

# ContextProvider 

Global data control component, used to replace global data.

======

> Props

|Property|Description|Type|DefaultValue|
|----------|-------------|------|------|
|prefixCls|Component classname prefix|string|"arco"|
|system|Manually control the current system, and the incoming value will be used directly after being passed in\. It is applicable when the initial value of the system needs to be specified in the ssr scenario\.|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|Whether to use dark mode|boolean|false|
|isDarkMode|Whether it is in dark mode|boolean|false|
|theme|Theme variable\. The css variable will be replaced online after input\. The less variable needs to be set|Record\<string, string\>|-|
|locale|Internationalized language configuration|ILocale|-|
|useRtl|Whether to use rtl|boolean|false|

> ILocale

|Property|Description|Type|
|----------|-------------|------|
|locale|Language Type|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
