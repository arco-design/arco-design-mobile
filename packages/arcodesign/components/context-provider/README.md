### 其他

# 全局配置 ContextProvider

全局数据控制组件，用于替换全局数据。

======

> 属性/Props

|参数|描述|类型|默认值|
|----------|-------------|------|------|
|prefixCls|组件类名前缀|string|"arco"|
|system|手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|是否使用暗黑模式|boolean|false|
|isDarkMode|是否处于暗黑模式，指定后以指定的值为准|boolean|false|
|theme|主题变量，传入后将在线替换css变量，需设置less变量 @use\-css\-vars: 1|Record\<string, string\>|-|
|locale|国际化语言包配置|ILocale|-|
|useRtl|是否使用Rtl模式|boolean|false|

> ILocale

|参数|描述|类型|
|----------|-------------|------|
|locale|语言类型|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
