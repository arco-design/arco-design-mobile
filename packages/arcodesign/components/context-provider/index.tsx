import React, { createContext, useState, useEffect, useCallback } from 'react';
import { addCssRules, removeCssStyleDom, defaultLocale, ILocale } from '@arco-design/mobile-utils';

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
    /**
     * 手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用
     * @en Manually control the current system, and the incoming value will be used directly after being passed in. It is applicable when the initial value of the system needs to be specified in the ssr scenario.
     * @default ""
     */
    system?: 'pc' | 'android' | 'ios' | '';
    /**
     * 是否使用暗黑模式
     * @en Whether to use dark mode
     * @default false
     */
    useDarkMode?: boolean;
    /**
     * 是否处于暗黑模式，指定后以指定的值为准
     * @en Whether it is in dark mode
     * @default false
     */
    isDarkMode?: boolean;
    /**
     * 主题变量，传入后将在线替换css变量，需设置less变量 @use-css-vars: 1
     * @en Theme variable. The css variable will be replaced online after input. The less variable needs to be set @use-css-vars: 1
     */
    theme?: Record<string, string>;
    /**
     * 国际化语言包配置
     * @en Internationalized language configuration
     */
    locale?: ILocale;
    /**
     * 是否使用Rtl模式
     * @en Whether to use rtl
     * @default false
     */
    useRtl?: boolean;
}

export const defaultContext: GlobalContextParams = {
    prefixCls: 'arco',
    system: '',
    useDarkMode: false,
    isDarkMode: false,
    locale: defaultLocale,
    useRtl: false,
};

export const GlobalContext = createContext<GlobalContextParams>(defaultContext);

export interface ContextProviderProps extends GlobalContextParams {
    children: React.ReactNode;
}

export type WithGlobalContext<T> = T & { context?: GlobalContextParams };

/**
 * 全局数据控制组件，用于替换全局数据。
 * @en Global data control component, used to replace global data.
 * @type 其他
 * @type_en Others
 * @name 全局配置
 * @name_en ContextProvider
 */
export default function ContextProvider(props: ContextProviderProps) {
    const {
        children,
        useDarkMode,
        isDarkMode,
        theme,
        locale = defaultLocale,
        ...restProps
    } = props;
    const [isDarkModeState, setIsDarkModeState] = useState<boolean>();
    const changeDarkMode = useCallback(
        (res: MediaQueryListEvent) => setIsDarkModeState(res.matches),
        [],
    );

    useEffect(() => {
        const matchMedia = window.matchMedia('(prefers-color-scheme: dark)');
        if (useDarkMode) {
            const dark = matchMedia.matches;
            setIsDarkModeState(dark);
            if (typeof matchMedia.addEventListener === 'function') {
                matchMedia.addEventListener('change', changeDarkMode);
            } else if (typeof matchMedia.addListener === 'function') {
                matchMedia.addListener(changeDarkMode);
            }
        }
        return () => {
            if (useDarkMode) {
                if (typeof matchMedia.removeEventListener === 'function') {
                    matchMedia.removeEventListener('change', changeDarkMode);
                } else if (typeof matchMedia.removeListener === 'function') {
                    matchMedia.removeListener(changeDarkMode);
                }
            }
        };
    }, [useDarkMode]);

    useEffect(() => {
        if (!theme) {
            return;
        }
        addCssRules('arcoCustomTheme', theme);
        return () => {
            removeCssStyleDom('arcoCustomTheme');
        };
    }, [theme]);

    return (
        <GlobalContext.Provider
            value={{
                ...defaultContext,
                ...restProps,
                isDarkMode: isDarkMode === void 0 ? isDarkModeState : isDarkMode,
                useDarkMode,
                locale: locale || defaultLocale,
            }}
        >
            {children}
        </GlobalContext.Provider>
    );
}

export const ContextLayout = GlobalContext.Consumer;

export function CompWithGlobalContext<P extends JSX.IntrinsicAttributes>(
    Component: React.FunctionComponent<P>,
) {
    return function (props: WithGlobalContext<P>) {
        const { context: propsContext, ...others } = props;
        return (
            <ContextLayout>
                {context => (
                    <ContextProvider {...{ ...context, ...propsContext }}>
                        <Component {...(others as P)} />
                    </ContextProvider>
                )}
            </ContextLayout>
        );
    };
}
