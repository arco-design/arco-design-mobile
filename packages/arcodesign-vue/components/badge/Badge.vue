<script lang="ts" setup>
import { getPrefixCls } from '../context-provider';

const prefix = getPrefixCls('badge');
</script>

<template>
    <button :class="prefix">按钮</button>
</template>
