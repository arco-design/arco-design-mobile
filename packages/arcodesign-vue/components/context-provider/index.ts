import ContextProvider from './ContextProvider.vue';
import { installFactory } from '../_helpers/install';

export type * from './type';
export * from './utils';

const install = installFactory(ContextProvider);

export { ContextProvider };

export default {
    install,
};
