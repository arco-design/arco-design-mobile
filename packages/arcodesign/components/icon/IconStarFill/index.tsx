import React, { SVGAttributes, CSSProperties } from 'react';
import { ContextLayout } from '../../context-provider';

export interface IconProps extends SVGAttributes<SVGElement> {
    className?: string;
    useCurrentColor?: boolean;
    style?: CSSProperties;
}

export default function IconStarFill(props: IconProps) {
    const { className = '', useCurrentColor = true, style, ...other } = props;

    return (
        <ContextLayout>
            {({ prefixCls }) => (
                <svg
                    className={`${prefixCls}-icon ${prefixCls}-icon-star-fill ${className}`}
                    width="1em"
                    height="1em"
                    style={style}
                    fill={useCurrentColor ? 'currentColor' : '#000'}
                    viewBox="0 0 1024 1024"
                    xmlns="http://www.w3.org/2000/svg"
                    {...other}
                >
                    <defs />
                    <path d="M497.371 826.514L270.63 936.23c-18.286 7.314-40.229 0-47.543-18.286-3.657-3.657-7.315-10.972-3.657-18.286l25.6-248.686c0-10.971-3.658-18.285-7.315-29.257L69.486 435.2c-14.629-14.629-10.972-36.571 3.657-51.2 3.657-3.657 10.971-7.314 14.628-7.314l245.029-51.2c10.971-3.657 18.286-7.315 21.943-18.286l128-219.429C490.057 73.143 512 65.83 530.286 76.8c3.657 3.657 10.971 7.314 14.628 14.629l128 219.428c3.657 7.314 14.629 14.629 21.943 18.286l245.029 51.2c18.285 3.657 32.914 21.943 25.6 43.886 0 7.314-3.657 10.971-7.315 14.628L789.943 621.714c-7.314 7.315-10.972 18.286-7.314 29.257l25.6 248.686c3.657 18.286-10.972 36.572-32.915 40.229-7.314 0-14.628 0-18.285-3.657L530.286 826.514c-10.972-7.314-25.6-7.314-32.915 0z" />
                </svg>
            )}
        </ContextLayout>
    );
}
