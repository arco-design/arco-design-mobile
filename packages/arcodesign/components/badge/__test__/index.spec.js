import React from 'react';
import { render } from '@testing-library/react';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import IconStarFill from '../../icon/IconStarFill';
import Badge from '..';
import { defaultContext } from '../../context-provider';

const prefix = `${defaultContext.prefixCls}-badge`;

demoTest('badge');

mountTest(Badge, 'Badge');

describe('Badge', () => {
    it('Render different types', () => {
        const { container: container1 } = render(<Badge absolute dot />);
        expect(container1.querySelectorAll(`.${prefix}.dot.absolute`).length).toBe(1);

        const { container: container2, getByText } = render(<Badge absolute text="12" />);
        expect(container2.querySelectorAll('.badge-text').length).toBe(1);
        expect(getByText('12')).toBeTruthy();

        const { container: container3, getByText: getByText99 } = render(
            <Badge absolute text="100" />,
        );
        expect(container3.querySelectorAll('.badge-text').length).toBe(1);
        expect(getByText99('99+')).toBeTruthy();

        const { container: container4, getByText: getByTextNew } = render(
            <Badge absolute text="新" />,
        );
        expect(container4.querySelectorAll('.badge-text').length).toBe(1);
        expect(getByTextNew('新')).toBeTruthy();

        const { container: container5 } = render(
            <Badge absolute>
                <IconStarFill />
            </Badge>,
        );
        expect(container5.querySelectorAll('svg').length).toBe(1);
    });
});
