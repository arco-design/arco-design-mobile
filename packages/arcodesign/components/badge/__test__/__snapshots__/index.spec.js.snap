// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`badge demo test badge demo: avatar.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="badge-demo-avatar-wrap"
  >
    <div
      class="badge-demo-avatar-box"
    >
      <div
        class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
      >
        <div
          class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
        >
          <div
            class="arco-image all-border-box pc arco-avatar-img"
          >
            <div
              class="image-content image-bottom-overlap"
            >
              <div
                class="image-placeholder"
              />
            </div>
            <div
              class="image-container loading animate"
              style="transition-duration: 200ms;"
            />
          </div>
        </div>
      </div>
      <div
        class="arco-badge arco-badge-dot dot arco-badge-bordered bordered arco-badge-absolute absolute"
        style="margin-left: -8px; margin-top: 4px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`badge demo test badge demo: define.md renders correctly 1`] = `
<DocumentFragment>
  <div
    style="display: flex; justify-content: space-around; position: relative; padding: 0.32rem 0px;"
  >
    <div
      class="badge-demo-item"
    >
      <div
        class="badge-demo-rectangle"
      >
        <div
          class="arco-badge arco-badge-dot dot arco-badge-absolute absolute"
          style="background: rgb(255, 87, 34);"
        />
      </div>
      <div
        class="badge-demo-item-text"
      >
        Custom Color
      </div>
    </div>
    <div
      class="badge-demo-item"
    >
      <div
        class="badge-demo-rectangle"
      >
        <div
          class="arco-badge arco-badge-absolute absolute"
          style="border-radius: 8px 8px 8px 0;"
        >
          <span
            class="arco-badge-text badge-text"
          >
            New
          </span>
        </div>
      </div>
      <div
        class="badge-demo-item-text"
      >
        Custom style
      </div>
    </div>
    <div
      class="badge-demo-item"
    >
      <div
        class="badge-demo-rectangle"
      >
        <div
          class="arco-badge arco-badge-absolute absolute"
        >
          <span
            class="arco-badge-text badge-text"
          >
            Gradient
          </span>
        </div>
      </div>
      <div
        class="badge-demo-item-text"
      >
        Custom color
      </div>
    </div>
    <div
      class="badge-demo-item"
    >
      <div
        class="badge-demo-rectangle"
      >
        <div
          class="arco-badge arco-badge-absolute absolute"
          style="background: rgb(255, 87, 34);"
        >
          <span
            class="arco-badge-text badge-text"
          >
            2
          </span>
        </div>
      </div>
      <div
        class="badge-demo-item-text"
      >
        Custom color
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`badge demo test badge demo: index.md renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="badge-demo-item-wrap"
    >
      <div
        style="display: flex; justify-content: space-around; position: relative;"
      >
        <div
          class="badge-demo-item"
        >
          <div
            class="badge-demo-rectangle"
          >
            <div
              class="arco-badge arco-badge-dot dot arco-badge-absolute absolute"
            />
          </div>
          <div
            class="badge-demo-item-text"
          >
            Dot badge
          </div>
        </div>
        <div
          class="badge-demo-item"
        >
          <div
            class="badge-demo-rectangle"
          >
            <div
              class="arco-badge arco-badge-absolute absolute"
            >
              <span
                class="arco-badge-text badge-text"
              >
                12
              </span>
            </div>
          </div>
          <div
            class="badge-demo-item-text"
          >
            Number badge
          </div>
        </div>
        <div
          class="badge-demo-item"
        >
          <div
            class="badge-demo-rectangle"
          >
            <div
              class="arco-badge arco-badge-absolute absolute"
            >
              <span
                class="arco-badge-text badge-text"
              >
                2
              </span>
            </div>
          </div>
          <div
            class="badge-demo-item-text"
          >
            Odd badge
          </div>
        </div>
        <div
          class="badge-demo-item"
        >
          <div
            class="badge-demo-rectangle"
          >
            <div
              class="arco-badge arco-badge-absolute absolute"
            >
              <span
                class="arco-badge-text badge-text"
              >
                99+
              </span>
            </div>
          </div>
          <div
            class="badge-demo-item-text"
          >
            Maximum display
          </div>
        </div>
      </div>
    </div>
    <div
      class="badge-demo-item-wrap"
    >
      <div
        style="display: flex; justify-content: space-around; position: relative;"
      >
        <div
          class="badge-demo-item"
        >
          <div
            class="badge-demo-rectangle"
          >
            <div
              class="arco-badge arco-badge-absolute absolute"
            >
              <span
                class="arco-badge-text badge-text"
              >
                New
              </span>
            </div>
          </div>
          <div
            class="badge-demo-item-text"
          >
            Text badge
          </div>
        </div>
        <div
          class="badge-demo-item"
        >
          <div
            class="badge-demo-rectangle"
          >
            <div
              class="arco-badge arco-badge-absolute absolute"
            >
              <span
                class="arco-badge-text badge-text"
              >
                On Sale
              </span>
            </div>
          </div>
          <div
            class="badge-demo-item-text"
          >
            Multiple text
          </div>
        </div>
        <div
          class="badge-demo-item"
        >
          <div
            class="badge-demo-rectangle"
          >
            <div
              class="arco-badge arco-badge-absolute absolute"
              style="padding: 0px; width: 16px; height: 16px;"
            >
              <svg
                class="arco-icon arco-icon-star-fill demo-icon"
                fill="currentColor"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs />
                <path
                  d="M497.371 826.514L270.63 936.23c-18.286 7.314-40.229 0-47.543-18.286-3.657-3.657-7.315-10.972-3.657-18.286l25.6-248.686c0-10.971-3.658-18.285-7.315-29.257L69.486 435.2c-14.629-14.629-10.972-36.571 3.657-51.2 3.657-3.657 10.971-7.314 14.628-7.314l245.029-51.2c10.971-3.657 18.286-7.315 21.943-18.286l128-219.429C490.057 73.143 512 65.83 530.286 76.8c3.657 3.657 10.971 7.314 14.628 14.629l128 219.428c3.657 7.314 14.629 14.629 21.943 18.286l245.029 51.2c18.285 3.657 32.914 21.943 25.6 43.886 0 7.314-3.657 10.971-7.315 14.628L789.943 621.714c-7.314 7.315-10.972 18.286-7.314 29.257l25.6 248.686c3.657 18.286-10.972 36.572-32.915 40.229-7.314 0-14.628 0-18.285-3.657L530.286 826.514c-10.972-7.314-25.6-7.314-32.915 0z"
                />
              </svg>
            </div>
          </div>
          <div
            class="badge-demo-item-text"
          >
            Icon badge
          </div>
        </div>
        <div
          class="badge-demo-item"
        >
          <div
            class="badge-demo-rectangle"
          >
            <div
              class="arco-badge arco-badge-absolute absolute"
            >
              <span
                class="arco-badge-text badge-text"
              >
                New
              </span>
            </div>
          </div>
          <div
            class="badge-demo-item-text"
          >
            Title text
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`badge demo test badge demo: list.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-cell-group all-border-box"
  >
    <div
      class="cell-group-body bordered"
    >
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              List content
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              class="arco-badge arco-badge-dot dot"
            />
          </div>
          <div
            class="cell-arrow-icon"
          >
            <svg
              class="arrow-icon-svg"
              fill="none"
              viewBox="0 0 8 14"
            >
              <path
                clip-rule="evenodd"
                d="M0.594858 1.24219C0.399596 1.43745 0.399596 1.75403 0.594859 1.94929L5.59905 6.95348L0.636039 11.9165C0.440776 12.1118 0.440776 12.4283 0.636038 12.6236L0.989592 12.9771C1.18485 13.1724 1.50144 13.1724 1.6967 12.9771L7.35355 7.3203C7.5296 7.14425 7.54692 6.86959 7.40553 6.67413C7.38216 6.62774 7.35111 6.58423 7.31237 6.54549L1.65552 0.888634C1.46026 0.693372 1.14367 0.693372 0.948411 0.888634L0.594858 1.24219Z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              List content
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              class="arco-badge"
            >
              <span
                class="arco-badge-text badge-text"
              >
                2
              </span>
            </div>
          </div>
          <div
            class="cell-arrow-icon"
          >
            <svg
              class="arrow-icon-svg"
              fill="none"
              viewBox="0 0 8 14"
            >
              <path
                clip-rule="evenodd"
                d="M0.594858 1.24219C0.399596 1.43745 0.399596 1.75403 0.594859 1.94929L5.59905 6.95348L0.636039 11.9165C0.440776 12.1118 0.440776 12.4283 0.636038 12.6236L0.989592 12.9771C1.18485 13.1724 1.50144 13.1724 1.6967 12.9771L7.35355 7.3203C7.5296 7.14425 7.54692 6.86959 7.40553 6.67413C7.38216 6.62774 7.35111 6.58423 7.31237 6.54549L1.65552 0.888634C1.46026 0.693372 1.14367 0.693372 0.948411 0.888634L0.594858 1.24219Z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="arco-cell all-border-box bordered"
      >
        <div
          class="arco-cell-inner"
        >
          <div
            class="cell-label"
          >
            <div
              class="cell-title"
            >
              List content
            </div>
          </div>
          <div
            class="cell-content has-label"
          >
            <div
              class="arco-badge"
            >
              <span
                class="arco-badge-text badge-text"
              >
                99+
              </span>
            </div>
          </div>
          <div
            class="cell-arrow-icon"
          >
            <svg
              class="arrow-icon-svg"
              fill="none"
              viewBox="0 0 8 14"
            >
              <path
                clip-rule="evenodd"
                d="M0.594858 1.24219C0.399596 1.43745 0.399596 1.75403 0.594859 1.94929L5.59905 6.95348L0.636039 11.9165C0.440776 12.1118 0.440776 12.4283 0.636038 12.6236L0.989592 12.9771C1.18485 13.1724 1.50144 13.1724 1.6967 12.9771L7.35355 7.3203C7.5296 7.14425 7.54692 6.86959 7.40553 6.67413C7.38216 6.62774 7.35111 6.58423 7.31237 6.54549L1.65552 0.888634C1.46026 0.693372 1.14367 0.693372 0.948411 0.888634L0.594858 1.24219Z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`badge demo test badge demo: toggle.md renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="arco-cell-group all-border-box"
    >
      <div
        class="cell-group-body bordered"
      >
        <div
          class="arco-cell all-border-box bordered"
        >
          <div
            class="arco-cell-inner"
          >
            <div
              class="cell-label"
            >
              <div
                class="cell-title"
              >
                <div
                  style="position: relative;"
                >
                  <span
                    class="badge-demo-cell-label"
                  >
                    List content
                    <div
                      class="arco-badge arco-badge-dot dot arco-badge-absolute absolute"
                      style="margin-left: 0px;"
                    />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="cell-content has-label"
            />
            <div
              class="cell-arrow-icon"
            >
              <svg
                class="arrow-icon-svg"
                fill="none"
                viewBox="0 0 8 14"
              >
                <path
                  clip-rule="evenodd"
                  d="M0.594858 1.24219C0.399596 1.43745 0.399596 1.75403 0.594859 1.94929L5.59905 6.95348L0.636039 11.9165C0.440776 12.1118 0.440776 12.4283 0.636038 12.6236L0.989592 12.9771C1.18485 13.1724 1.50144 13.1724 1.6967 12.9771L7.35355 7.3203C7.5296 7.14425 7.54692 6.86959 7.40553 6.67413C7.38216 6.62774 7.35111 6.58423 7.31237 6.54549L1.65552 0.888634C1.46026 0.693372 1.14367 0.693372 0.948411 0.888634L0.594858 1.24219Z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="arco-cell all-border-box bordered"
        >
          <div
            class="arco-cell-inner"
          >
            <div
              class="cell-label"
            >
              <div
                class="cell-title"
              >
                <div
                  style="position: relative;"
                >
                  <span
                    class="badge-demo-cell-label"
                  >
                    List content
                    <div
                      class="arco-badge arco-badge-absolute absolute"
                      style="margin-left: 2px;"
                    >
                      <span
                        class="arco-badge-text badge-text"
                      >
                        2
                      </span>
                    </div>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="cell-content has-label"
            />
            <div
              class="cell-arrow-icon"
            >
              <svg
                class="arrow-icon-svg"
                fill="none"
                viewBox="0 0 8 14"
              >
                <path
                  clip-rule="evenodd"
                  d="M0.594858 1.24219C0.399596 1.43745 0.399596 1.75403 0.594859 1.94929L5.59905 6.95348L0.636039 11.9165C0.440776 12.1118 0.440776 12.4283 0.636038 12.6236L0.989592 12.9771C1.18485 13.1724 1.50144 13.1724 1.6967 12.9771L7.35355 7.3203C7.5296 7.14425 7.54692 6.86959 7.40553 6.67413C7.38216 6.62774 7.35111 6.58423 7.31237 6.54549L1.65552 0.888634C1.46026 0.693372 1.14367 0.693372 0.948411 0.888634L0.594858 1.24219Z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="arco-cell all-border-box bordered"
        >
          <div
            class="arco-cell-inner"
          >
            <div
              class="cell-label"
            >
              <div
                class="cell-title"
              >
                <div
                  style="position: relative;"
                >
                  <span
                    class="badge-demo-cell-label"
                  >
                    List content
                    <div
                      class="arco-badge arco-badge-absolute absolute"
                      style="margin-left: 2px;"
                    >
                      <span
                        class="arco-badge-text badge-text"
                      >
                        99+
                      </span>
                    </div>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="cell-content has-label"
            />
            <div
              class="cell-arrow-icon"
            >
              <svg
                class="arrow-icon-svg"
                fill="none"
                viewBox="0 0 8 14"
              >
                <path
                  clip-rule="evenodd"
                  d="M0.594858 1.24219C0.399596 1.43745 0.399596 1.75403 0.594859 1.94929L5.59905 6.95348L0.636039 11.9165C0.440776 12.1118 0.440776 12.4283 0.636038 12.6236L0.989592 12.9771C1.18485 13.1724 1.50144 13.1724 1.6967 12.9771L7.35355 7.3203C7.5296 7.14425 7.54692 6.86959 7.40553 6.67413C7.38216 6.62774 7.35111 6.58423 7.31237 6.54549L1.65552 0.888634C1.46026 0.693372 1.14367 0.693372 0.948411 0.888634L0.594858 1.24219Z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
      type="button"
    >
      <div
        class="arco-button-text arco-button-text-pc btn-text"
      >
        Close Badge
      </div>
    </button>
  </div>
</DocumentFragment>
`;
